lhlist/src/ordered_set.rs

```diff
@@ -0,0 +1,103 @@
+use crate::Label;
+use crate::{Cons, False, LabeledValue, Member, Nil};
+
+impl OrderedHSet for Nil {}
+
+impl<H, T> OrderedHSet for Cons<LabeledValue<H>, T>
```

Comment:
Does this need to be `Cons<LabeledValue<H>, T>` or can it be `Cons<H, T>` where `H` is anything that implements `Label`? The `Member` trait is implemented for anything that implements `Label`, not just `LabeledValue`, and will implement `OrderedHSet` for `LCons` (cons-lists that only contains `PhantomData` label information).

Reply:
Also, elsewhere I typically use `Cons<L, T> where L: Label` instead of `Cons<H, T>` when working with labeled cons-lists, but that's not strictly necessary

Reply:
You are right, we don't need it to be a `LabeledValue` - changed it to be more general :+1: